File: cse3302/rxm7694_lab01.java
//Rodrigo Munoz
//ID: 1001847694
//macos 13.3.1 (a) (22E772610a)
//java 18.0.1.1 2022-04-22

import java.io.File;  // Import the File class
import java.io.IOException;  // Import the IOException class to handle errors

public class rxm7694_lab01{
    public static void main(String[] args) throws IOException{
      long dirSize = getDirSize("./");

      System.out.println("Dir size = " + dirSize);
    }

    public static long getDirSize(String path) throws IOException {
        long size = 0;
        File[] dir = new File(path).listFiles();

        for(File file : dir){
            if(file.isFile()){
                size += file.length();
            }
            else if(file.isDirectory() && file.getName() != "." && file.getName() != ".."){
                size = size + getDirSize(file.getPath());
            }
        }

        return size;
    }
}

File: cse3302/rxm7694_lab01.pl
#Rodrigo Munoz
#ID: 1001847694
#macos 13.3.1 (a) (22E772610a)
#This is perl 5, version 30, subversion 3 (v5.30.3) built for darwin-thread-multi-2level

#!/usr/bin/perl
use strict;
use warnings;

sub dirSize{
    my ($path) = @_;
    my $size = 0;

    #open the current working directory
    opendir(my $dir, $path) || die "Failed to open: $path\n";

    # iterate through files and directories
    while (my $file = readdir($dir)) {
        # check for current and prev dir
        next if ($file eq '.' || $file eq '..');
        my $newPath = "$path/$file";
        # if file
        if (-f $newPath) {
            $size += -s $newPath;
        }
        # if dir
        elsif (-d $newPath) {
            $size += dirSize($newPath);
        }
    }

    #clean up
    closedir($dir);
    return $size;
}

my $dirPath = '.';
my $size = dirSize($dirPath);

print "Dir size = $size\n";
